### PR TITLE
 Add basic functions for change of basis in 3D

### PR DESCRIPTION
**Pull request by oOKexOo**
**When merged this pull request will:**
- Add `CBA_fnc_matrixProduct3D` for the product of two 3x3-matrices.
- Add `CBA_fnc_matrixTranspose` for transposing a mxn-matrix.
- Add `CBA_fnc_vectMap3D` for 3x3-matrix - 3D-vector multiplication.

### CODE REVIEW

File: addons/vectors/fnc_matrixTranspose.sqf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+private _m = count _matrix;
+private _n = count (_matrix select 0);
+
+private _returnMatrix = [];
+
+for "_j" from 0 to (_n - 1) do
+{
+    private _returnRow = [];
+    for "_i" from 0 to (_m - 1) do
```
Comment (dedmen) [inline addons/vectors/fnc_matrixTranspose.sqf]: use `apply` here?

Comment (oOKexOo) [reply addons/vectors/fnc_matrixTranspose.sqf]: Agreed, the same goes CBA_fnc_matrixProduct3D.

**Comment by commy2** [comment]
Made some ocd style fixes.
These probably should have [unit tests here](https://github.com/CBATeam/CBA_A3/blob/master/addons/vectors/test_vectors.sqf#L297).

File: addons/vectors/test_vectors.sqf [hunk 2]
```diff
@@ -12,13 +12,28 @@ private _fnc_vectorEquals = {
     if ((count _vector1) != (count _vector2)) exitWith {false};
     private _equal = true;
     {
-        if ((abs (_x - (_vector2 select _forEachIndex))) > 0.00001) then {
```
Comment (dedmen) [inline addons/vectors/test_vectors.sqf]: can be replaced by findIf I think.

Comment (oOKexOo) [reply addons/vectors/test_vectors.sqf]: `findif` does not support `_forEachIndex`

Comment (commy2) [reply addons/vectors/test_vectors.sqf]: No need to, findIf is kind of hard to read.

**Comment by commy2** [comment]
```sqf
execVM "\x\cba\addons\vectors\test_vectors.sqf"
```

```sqf
 9:31:27 [CBA] (vectors) LOG: Testing Vectors x\cba\addons\vectors\test_vectors.sqf:39
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_polar2vect is defined) x\cba\addons\vectors\test_vectors.sqf:43
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:47
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:51
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:55
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:59
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_scaleVect is defined) x\cba\addons\vectors\test_vectors.sqf:64
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:68
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:72
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:76
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_scaleVectTo is defined) x\cba\addons\vectors\test_vectors.sqf:81
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:85
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:89
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:93
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:97
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_simplifyAngle is defined) x\cba\addons\vectors\test_vectors.sqf:102
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_simplifyAngle x\cba\addons\vectors\test_vectors.sqf:106
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_simplifyAngle x\cba\addons\vectors\test_vectors.sqf:110
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_simplifyAngle x\cba\addons\vectors\test_vectors.sqf:114
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_simplifyAngle180 is defined) x\cba\addons\vectors\test_vectors.sqf:119
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_simplifyAngle180 x\cba\addons\vectors\test_vectors.sqf:123
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_simplifyAngle180 x\cba\addons\vectors\test_vectors.sqf:127
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_simplifyAngle180 x\cba\addons\vectors\test_vectors.sqf:131
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vect2Polar is defined) x\cba\addons\vectors\test_vectors.sqf:136
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:140
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:144
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:148
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:152
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectAdd is defined) x\cba\addons\vectors\test_vectors.sqf:158
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:162
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:166
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectCross is defined) x\cba\addons\vectors\test_vectors.sqf:172
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:176
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectCross2D is defined) x\cba\addons\vectors\test_vectors.sqf:181
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectCross2D x\cba\addons\vectors\test_vectors.sqf:185
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectDir is defined) x\cba\addons\vectors\test_vectors.sqf:190
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectDir x\cba\addons\vectors\test_vectors.sqf:194
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectDir x\cba\addons\vectors\test_vectors.sqf:198
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectDot is defined) x\cba\addons\vectors\test_vectors.sqf:203
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectDot x\cba\addons\vectors\test_vectors.sqf:207
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectDot x\cba\addons\vectors\test_vectors.sqf:211
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectElev is defined) x\cba\addons\vectors\test_vectors.sqf:216
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectElev x\cba\addons\vectors\test_vectors.sqf:220
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectElev x\cba\addons\vectors\test_vectors.sqf:224
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectMagn is defined) x\cba\addons\vectors\test_vectors.sqf:229
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectMagn x\cba\addons\vectors\test_vectors.sqf:233
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectMagn x\cba\addons\vectors\test_vectors.sqf:237
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectMagn2D is defined) x\cba\addons\vectors\test_vectors.sqf:242
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectMagn2D x\cba\addons\vectors\test_vectors.sqf:246
 9:31:27 [CBA] (vectors) Test OK: (_result == _expected) CBA_fnc_vectMagn2D x\cba\addons\vectors\test_vectors.sqf:250
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectRotate2D is defined) x\cba\addons\vectors\test_vectors.sqf:255
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:259
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:263
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectRotate3D is defined) x\cba\addons\vectors\test_vectors.sqf:267
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:271
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:275
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:279
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:283
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:287
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:291
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectSubtract is defined) x\cba\addons\vectors\test_vectors.sqf:295
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:299
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:306
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:311
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_vectMap3D is defined) x\cba\addons\vectors\test_vectors.sqf:316
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:320
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_vectorEquals) x\cba\addons\vectors\test_vectors.sqf:324
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_matrixTranspose is defined) x\cba\addons\vectors\test_vectors.sqf:328
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_matrixEquals) x\cba\addons\vectors\test_vectors.sqf:332
 9:31:27 [CBA] (vectors) Test OK: (CBA_fnc_matrixProduct3D is defined) x\cba\addons\vectors\test_vectors.sqf:336
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_matrixEquals) x\cba\addons\vectors\test_vectors.sqf:340
 9:31:27 [CBA] (vectors) Test OK: ([_result, _expected] call _fnc_matrixEquals) x\cba\addons\vectors\test_vectors.sqf:346
```

**Pull request merged**